src/libs/actions/Report.js

```diff
@@ -319,6 +320,44 @@ function fetchIOUReportID(debtorEmail) {
     });
 }
 
+function configureReportNameAndIcon(reports) {
+    // The personalDetails of the participants contain their avatar images. Here we'll go over each
+    // report and based on the participants we'll link up their avatars to report icons. This will
+    // skip over default rooms which aren't named by participants.
+
+    Onyx.connect({
+        key: ONYXKEYS.PERSONAL_DETAILS,
+        callback: (formattedPersonalDetails) => {
+            const reportsToUpdate = {};
```

Comment:
Where are you canceling the subscription?

Reply:
I didn't find anything about cancelling the subscription in the code as well as documentation.
Do you mean?
```diff
    Onyx.connect({
        key: ONYXKEYS.PERSONAL_DETAILS,
        callback: (formattedPersonalDetails) => {

+            if (!formattedPersonalDetails) {
+               return;
+          }

            const reportsToUpdate = {};
```

Reply:
Yeah, that is a good idea. 

> Action methods should only have return values (data or a promise) if they are called by other actions. This is done to encourage that action methods can be called in parallel with no dependency on other methods (see discussion above).